packages/dev/core/src/FlowGraph/flowGraphExecutionBlock.ts

```diff
@@ -0,0 +1,41 @@
+import type { FlowGraph } from "./flowGraph";
+import { FlowGraphBlock } from "./flowGraphBlock";
+import { FlowGraphConnectionPointDirection, FlowGraphSignalConnectionPoint } from "./flowGraphConnectionPoint";
+
+/**
+ * @experimental
+ * A block that executes some action. Always has an input signal.
+ * Can have one or more output signals.
+ */
+export abstract class FlowGraphExecutionBlock extends FlowGraphBlock {

```

Comment:
should take async execution into account. A thought for the future is whether or not an async execution can be canceled while executed, and what does it mean to the architecture